src/components/MultiStepForm/MultiStepForm.tsx

```diff
@@ -38,7 +40,25 @@ const FormContainer: FC<MultiStepFormProps> = ({ formTemplate, formData, isLoadi
     }
   }, [header, pageNumber]);
 
-  const onButtonClick = (): void => {
+  const onButtonClick = (e: FormEvent<Element>): void => {
+    e.preventDefault();
+
+    const errors: Record<string, string> = {};
+    for (const component of formComponents) {
```

Comment:
This would be better as a reduce, so something like:

const errors = `Object.keys(formComponents).reduce((acc, {componentData}) => {`
      `const { formMeta: { field = '' } = {} } = componentData as CommonInputProps;`
      ` const error = validateFormInputField(formData, field);`
      `if (error) {`
          `acc[field] = error;`
      `}`
      `return acc;`
  `}, {});`

Reply:
good suggestion, I've now implemented this